crates/turbo-tasks-memory/src/task.rs

```diff
@@ -628,49 +604,72 @@ impl Task {
             // once task won't become dirty
             return;
         }
-        self.clear_dependencies(backend);
 
-        let mut state = self.state.write();
-        match state.state_type {
-            Dirty | Scheduled | InProgressDirty => {
-                // already dirty
-            }
-            Done => {
-                // add to dirty lists and potentially schedule
-                let mut active = false;
-                for scope in state.scopes.iter() {
-                    backend.with_scope(scope, |scope| {
-                        scope.increment_unfinished_tasks(backend);
-                        log_scope_update!("add unfinished task: {} -> {}", *scope.id, *self.id);
-                        let mut scope = scope.state.lock();
-                        if scope.is_active() {
-                            active = true;
-                        } else {
-                            scope.add_dirty_task(self.id);
-                        }
-                    });
-                }
-                if active {
-                    state.state_type = Scheduled;
+        let id = self.id;
+        let mut clear_dependencies = None;
```

Comment:
This could be an empty vec, since empty vecs don't allocate.

```suggestion
        let mut clear_dependencies = vec![];
```

Reply:
It's an HashSet, but it will change to an AutoSet soon, where it would be a Vec